Sources/xcproj/PBXProj.swift

```diff
@@ -101,48 +101,46 @@ final public class PBXProj: Decodable {
         }
 
         public func getTarget(reference: String) -> PBXTarget? {
-            let caches: [[String: PBXTarget]] = [
-                aggregateTargets,
-                nativeTargets,
-                legacyTargets
-            ]
-            return caches.first { cache in cache[reference] != nil }?[reference]
+            return aggregateTargets[reference] ??
+                nativeTargets[reference] ??
+                legacyTargets[reference]
         }
 
         public func getFileElement(reference: String) -> PBXFileElement? {
-            let caches: [[String: PBXFileElement]] = [
-                fileReferences,
-                groups,
-                variantGroups,
-                versionGroups,
-                ]
-            return caches.first { cache in cache[reference] != nil }?[reference]
+            return fileReferences[reference] ??
+                groups[reference] ??
+                variantGroups[reference] ??
+                versionGroups[reference]
         }
 
         public func getReference(_ reference: String) -> PBXObject? {
-            let caches: [[String: PBXObject]] = [
-                buildFiles,
-                aggregateTargets,
-                legacyTargets,
-                containerItemProxies,
-                groups,
-                configurationLists,
-                buildConfigurations,
-                variantGroups,
-                targetDependencies,
-                nativeTargets,
-                fileReferences,
-                projects,
-                versionGroups,
-                referenceProxies,
-                copyFilesBuildPhases,
-                shellScriptBuildPhases,
-                resourcesBuildPhases,
-                frameworksBuildPhases,
-                headersBuildPhases,
-                sourcesBuildPhases
-            ]
-            return caches.first { cache in cache[reference] != nil }?[reference]
+            // The ?? chaining below is partitioned into separate statments because, with Swift 4, a single
+            // expression resulted in this compiler error:
+            //     Expression was too complex to be solved in reasonable time;
```

Comment:
Out of curiosity is the error thrown when you try to build from Xcode or when you build from the console? @kastiglione

Reply:
Xcode. I haven't yet built from console.

Reply:
👍